src/DependenciesPackager/DependenciesPackager.csproj

```diff
@@ -18,4 +18,17 @@
     <PackageReference Include="System.Diagnostics.Process" Version="4.1.0" />
   </ItemGroup>
 
+  <!-- packaging settings -->
+  <PropertyGroup>
+    <NoPackageAnalysis>true</NoPackageAnalysis>
+    <NuspecFile>$(MSBuildThisFileDirectory)$(MSBuildProjectName).nuspec</NuspecFile>
+  </PropertyGroup>
+
+  <Target Name="SetPackageDependencies" BeforeTargets="GenerateNuspec" DependsOnTargets="Publish">
+    <PropertyGroup>
```

Comment:
You should have a property named SetPackageDependenciesDependsOn and do this through <Target Name="SetPackageDependencies" DependsOnTargets="$(SetPackageDependenciesDependsOn)">

Mixing BeforeTargets and DependsOnTargets is kind of ugly and using properties with DependsOnTargets allows you to:
1) Allow third party scripts to plug in transparently into your build script by just defining <Property>$(Property);newvalue</Property>
2) Allows you to see all the dependency graph of tasks in one place. (the property group where all these things are defined.
3) You don't have to be hunting around to modify the dependency graph.